panther_utils/panther_utils/arguments.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+# Copyright 2024 Husarion sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Text, Tuple
+
+import yaml
+from launch.action import Action
+from launch.actions import DeclareLaunchArgument
+from launch.launch_context import LaunchContext
+from launch.substitutions import EnvironmentVariable, LaunchConfiguration
+from launch.utilities import normalize_to_list_of_substitutions, perform_substitutions
+
+# Define valid configurations for each robot model
+VALID_CONFIGURATIONS = {
+    "panther": {"wheel_type": ["WH01", "WH02", "WH04"]},
+    "lynx": {"wheel_type": ["WH05"]},
+}
+
+
+class DeclareRobotArgs(Action):
+    """Retrieves and validate the robot configuration from the YAML data."""
+
+    def __init__(self, robot_configuration_path: LaunchConfiguration, **kwargs) -> None:
+        """Create a DeclareRobotArgs action."""
+        super().__init__(**kwargs)
+        self.__path = normalize_to_list_of_substitutions(robot_configuration_path)
```

Comment:
```suggestion
        self._path = normalize_to_list_of_substitutions(robot_configuration_path)
```
Don't think this will have any subclasses so single underscore should be just fine

Reply:
not actual